From 632a81f92a81f40b1d4b357006488891d53695f2 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <sanyog.r.kale@intel.com>
Date: Fri, 6 Jan 2017 15:31:24 +0530
Subject: [PATCH 4068/4301] SoundWire: Remove Maxim FPGA support from
 SoundWire bus

commit fc2f46b1ca02118a3345fb2cf5dca8c3459f17c9 from
git://git@github.com:01org/linux-intel-4.9.git

Maxim codec FPGA is no more used for SoundWire use case verification,
removing related code changes.

Change-Id: I7584e7f81922df3f3d168d41ef7192a6449ff044
Signed-off-by: Sanyog Kale <sanyog.r.kale@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/12935
Reviewed-by: Koul, Vinod <vinod.koul@intel.com>
Reviewed-by: Singh, Guneshwor O <guneshwor.o.singh@intel.com>
Reviewed-by: Nc, Shreyas <shreyas.nc@intel.com>
Reviewed-by: Diwakar, Praveen <praveen.diwakar@intel.com>
Tested-by: Avati, Santosh Kumar <santosh.kumar.avati@intel.com>
---
 drivers/sdw/sdw_bwcalc.c |   28 +++-------------------------
 1 files changed, 3 insertions(+), 25 deletions(-)

diff --git a/drivers/sdw/sdw_bwcalc.c b/drivers/sdw/sdw_bwcalc.c
index 14460cc..78e103f 100644
--- a/drivers/sdw/sdw_bwcalc.c
+++ b/drivers/sdw/sdw_bwcalc.c
@@ -247,17 +247,13 @@ int sdw_cfg_slv_params(struct sdw_bus *mstr_bs,
 	rd_msg.buf = rbuf;
 	rd_msg.addr_page1 = 0x0;
 	rd_msg.addr_page2 = 0x0;
-/* Dont program slave params for the Aggregation.
- * Its with master loop back
- */
-#ifndef CONFIG_SND_SOC_MXFPGA
+
 	ret = sdw_slave_transfer(mstr_bs->mstr, &rd_msg, 1);
 	if (ret != 1) {
 		ret = -EINVAL;
 		dev_err(&mstr_bs->mstr->dev, "Register transfer failed\n");
 		goto out;
 	}
-#endif
 
 	wbuf1[0] = (p_slv_params->port_flow_mode |
 			(p_slv_params->port_data_mode <<
@@ -296,10 +292,7 @@ int sdw_cfg_slv_params(struct sdw_bus *mstr_bs,
 	wr_msg1.buf = &wbuf1[0];
 	wr_msg1.addr_page1 = 0x0;
 	wr_msg1.addr_page2 = 0x0;
-/* Dont program slave params for the Aggregation.
- * Its with master loop back
- */
-#ifndef CONFIG_SND_SOC_MXFPGA
+
 	ret = sdw_slave_transfer(mstr_bs->mstr, &wr_msg, 1);
 	if (ret != 1) {
 		ret = -EINVAL;
@@ -315,7 +308,6 @@ int sdw_cfg_slv_params(struct sdw_bus *mstr_bs,
 		goto out;
 	}
 out:
-#endif
 
 	return ret;
 }
@@ -492,10 +484,6 @@ int sdw_cfg_slv_enable_disable(struct sdw_bus *mstr_bs,
 		 */
 
 		/* 2. slave port enable */
-/* Dont program slave params for the Aggregation.
- * Its with master loop back
- */
-#ifndef CONFIG_SND_SOC_MXFPGA
 		ret = sdw_slave_transfer(mstr_bs->mstr, &rd_msg, 1);
 		if (ret != 1) {
 			ret = -EINVAL;
@@ -522,7 +510,6 @@ int sdw_cfg_slv_enable_disable(struct sdw_bus *mstr_bs,
 					"Register transfer failed\n");
 			goto out;
 		}
-#endif
 		/*
 		 * 3. slave port enable post pre
 		 * --> callback
@@ -537,10 +524,6 @@ int sdw_cfg_slv_enable_disable(struct sdw_bus *mstr_bs,
 		 * --> callback
 		 * --> no callback available
 		 */
-/* Dont program slave params for the Aggregation.
- * Its with master loop back
- */
-#ifndef CONFIG_SND_SOC_MXFPGA
 
 		/* 2. slave port disable */
 		ret = sdw_slave_transfer(mstr_bs->mstr, &rd_msg, 1);
@@ -569,7 +552,7 @@ int sdw_cfg_slv_enable_disable(struct sdw_bus *mstr_bs,
 					"Register transfer failed\n");
 			goto out;
 		}
-#endif
+
 		/*
 		 * 3. slave port enable post unpre
 		 * --> callback
@@ -579,9 +562,7 @@ int sdw_cfg_slv_enable_disable(struct sdw_bus *mstr_bs,
 			slv_rt_strm->rt_state = SDW_STATE_DISABLE_RT;
 
 	}
-#ifndef CONFIG_SND_SOC_MXFPGA
 out:
-#endif
 	return ret;
 
 }
@@ -1585,9 +1566,6 @@ int sdw_cfg_slv_prep_unprep(struct sdw_bus *mstr_bs,
 	wr_msg.addr_page1 = 0x0;
 	wr_msg.addr_page2 = 0x0;
 
-#ifdef CONFIG_SND_SOC_MXFPGA
-	sdw_slv_dpn_cap->prepare_ch = 0;
-#endif
 	if (prep) { /* PREPARE */
 
 		/*
-- 
1.7.5.4

